src/network/peers.rs

```diff
@@ -61,6 +65,9 @@ type PeersHandler<N, E> = mpsc::Receiver<PeersRequest<N, E>>;
 /// Shorthand for the parent half of the connection result channel.
 type ConnectionResult = oneshot::Sender<Result<()>>;
 
+/// A cache containing fast hashes of recent blocks.
+static BLOCK_CACHE: Lazy<RwLock<CircularQueue<u64>>> = Lazy::new(|| RwLock::new(CircularQueue::with_capacity(512)));
```

Comment:
Just curious, why `512`?

Reply:
This is a somewhat arbitrary pick; it needs to be small enough to be super fast, but also large enough to filter out all recently seen blocks; it might require fine-tuning.